util/dvsim/utils.py

```diff
@@ -165,7 +165,7 @@ def _subst_wildcards(var, mdict, ignored, ignore_error, seen):
             # command and we're done.
             cmd_matches = list(wildcard_re.finditer(cmd))
             if not cmd_matches:
-                var = var[:idx] + run_cmd(cmd)
+                var = var[:match.start()] + run_cmd(cmd)
```

Comment:
Yikes, good catch! We've actually got some tests for this code in utils_test.py: could you add something to check this stays fixed? (Also, we should probably wire that test into CI: I'll add it to my todo list).

Reply:
Added in #5927.